src/Compilers/CSharp/Portable/BoundTree/BoundNodes.xml

```diff
@@ -1896,18 +1899,27 @@
     <Field Name="CollectionBuilderInvocationPlaceholder" Type="BoundValuePlaceholder?" SkipInVisitor="true" Null="allow"/>
     <!-- Conversion from collection builder method invocation to collection type. -->
     <Field Name="CollectionBuilderInvocationConversion" Type="BoundExpression?" SkipInVisitor="true" Null="allow"/>
-    <!-- Collection expression elements. -->
-    <Field Name="Elements" Type="ImmutableArray&lt;BoundExpression&gt;"/>
   </Node>
 
+  <Node Name="BoundCollectionExpressionSpreadExpressionPlaceholder" Base="BoundValuePlaceholderBase"/>
+
   <Node Name="BoundCollectionExpressionSpreadElement" Base="BoundExpression">
-    <!-- Non-null type is required for this node kind -->
-    <Field Name="Type" Type="TypeSymbol" Override="true" Null="disallow"/>
+    <!-- Type is not significant for this node type; always null -->
+    <Field Name="Type" Type="TypeSymbol?" Override="true" Null="always"/>
+    <!-- Collection being spread. -->
     <Field Name="Expression" Type="BoundExpression"/>
+    <!-- Collection placeholder, used in Conversion and LengthOrCount fields. -->
+    <Field Name="ExpressionPlaceholder" Type="BoundCollectionExpressionSpreadExpressionPlaceholder?" SkipInVisitor="true" Null="allow"/>
+    <!-- Collection conversion to the type expected for foreach. -->
+    <Field Name="Conversion" Type="BoundExpression?" Null="allow" SkipInVisitor="true"/>
+    <!-- Collection enumerator info. -->
     <Field Name="EnumeratorInfoOpt" Type="ForEachEnumeratorInfo?"/>
+    <!-- Collection Length or Count property value. -->
+    <Field Name="LengthOrCount" Type="BoundExpression?" SkipInVisitor="true" Null="allow"/>
+    <!-- Collection element placeholder. -->
```

Comment:
It's unclear to me from reading these comments what the difference between this and `ExpressionPlaceholder` is. Consider elaborating why we have both and what the difference between them is.

Reply:
`ExpressionPlaceholder` represents the entire collection and `ElementPlaceholder` represents the current item in the iteration. Added a comment where `ElementPlaceholder` is used.